homeassistant/components/media_player/onkyo.py

```diff
@@ -289,9 +293,18 @@ def update(self):
         current_source_raw = self.command(
             'zone{}.selector=query'.format(self._zone))
 
+        # If we received a source value, but not a volume value
+        # it's likely this zone permanently does not support volume.
+        if current_source_raw and not volume_raw:
+            self._supports_volume = False
+
         if not (volume_raw and mute_raw and current_source_raw):
             return
 
+        # It's possible for some players to have zones set to HDMI with
+        # no sound control. In this case, the string `N/A` is returned.
+        self._supports_volume = isinstance(volume_raw[1], (float, int))
```

Comment:
Interesting, hadn't seen that discussed. The features here are less driven by state (idle, on etc.) and more by configuration (set zone 2 to HDMI out instead of speaker out), so they can change during the lifetime of the entity, but aren't expected to do so regularly.

Reply:
The problem is that for example, a device is synced to Google Assistant or Amazon Echo exactly once. If it changes, things might sync over incorrectly.

Reply:
I think I can move the code determining this into the constructor (not my initial code) and that would avoid the problem? It would add a delay in initialising the entity since it needs to make network requests, I assume that's acceptable and handled correctly.